packages/git/src/browser/dirty-diff/dirty-diff-manager.ts

```diff
@@ -202,18 +205,19 @@ export class DirtyDiffModel implements Disposable {
             this.dirty = false;
             this.previousContent = [];
         }
-        if (noRelevantChanges && this.isInGitRepository(repository)) {
+        const inGitRepository = await this.isInGitRepository(repository);
```

Comment:
just a minor, but could we push the "expensive" one level deeper after evaluation of `noRelevantChanges`?